diff --git a/node_modules/@testing-library/react/dist/@testing-library/react.esm.js b/node_modules/@testing-library/react/dist/@testing-library/react.esm.js
index f25d5fb..e647f88 100644
--- a/node_modules/@testing-library/react/dist/@testing-library/react.esm.js
+++ b/node_modules/@testing-library/react/dist/@testing-library/react.esm.js
@@ -31,15 +31,17 @@ try {
   // we can't use regular timers because they may still be faked
   // so we try MessageChannel+postMessage instead
   enqueueTask = function (callback) {
-    if (didWarnAboutMessageChannel === false) {
+    const supportsMessageChannel = typeof MessageChannel === 'function';
+    if (supportsMessageChannel) {
+      var channel = new MessageChannel();
+      channel.port1.onmessage = callback;
+      channel.port2.postMessage(undefined);
+    } else if (didWarnAboutMessageChannel === false) {
       didWarnAboutMessageChannel = true; // eslint-disable-next-line no-console
 
-      console.error(typeof MessageChannel !== 'undefined', 'This browser does not have a MessageChannel implementation, ' + 'so enqueuing tasks via await act(async () => ...) will fail. ' + 'Please file an issue at https://github.com/facebook/react/issues ' + 'if you encounter this warning.');
+      console.error('This browser does not have a MessageChannel implementation, ' + 'so enqueuing tasks via await act(async () => ...) will fail. ' + 'Please file an issue at https://github.com/facebook/react/issues ' + 'if you encounter this warning.');
     }
 
-    var channel = new MessageChannel();
-    channel.port1.onmessage = callback;
-    channel.port2.postMessage(undefined);
   };
 }
 
