diff --git a/esm/module.js b/esm/module.js
index 527070166142e7fda5ed14e1f3465052c82f2a2b..122061e0d481066a2b5fd0c7ab2a8b1d523b1ef1 100644
--- a/esm/module.js
+++ b/esm/module.js
@@ -119,7 +119,7 @@ export class Module {
     this.services = services;
     this.moduleImpl = moduleImpl;
     this.cache = services.cache;
-    this.#entrypointRef = new WeakRef(entrypoint);
+    this.#entrypointRef = entrypoint;
     this.idx = entrypoint.idx;
     this.id = entrypoint.name;
     this.filename = entrypoint.name;
@@ -143,7 +143,7 @@ export class Module {
     this.debug('the whole exports was overridden with %O', value);
   }
   get entrypoint() {
-    const entrypoint = this.#entrypointRef.deref();
+    const entrypoint = this.#entrypointRef;
     invariant(entrypoint, `Module ${this.idx} is disposed`);
     return entrypoint;
   }
diff --git a/lib/module.js b/lib/module.js
index d2f2644c2469ac7ad0dbd1b9f58099eda4084b80..b8bad9c43c35bef14c3f6a277f8c9974607b2a00 100644
--- a/lib/module.js
+++ b/lib/module.js
@@ -127,7 +127,7 @@ class Module {
     this.services = services;
     this.moduleImpl = moduleImpl;
     this.cache = services.cache;
-    this.#entrypointRef = new WeakRef(entrypoint);
+    this.#entrypointRef = entrypoint;
     this.idx = entrypoint.idx;
     this.id = entrypoint.name;
     this.filename = entrypoint.name;
@@ -151,7 +151,7 @@ class Module {
     this.debug('the whole exports was overridden with %O', value);
   }
   get entrypoint() {
-    const entrypoint = this.#entrypointRef.deref();
+    const entrypoint = this.#entrypointRef;
     (0, _tsInvariant.invariant)(entrypoint, `Module ${this.idx} is disposed`);
     return entrypoint;
   }
