diff --git a/node_modules/argos-cli/lib/index.js b/node_modules/argos-cli/lib/index.js
index 3a56364..49754e4 100644
--- a/node_modules/argos-cli/lib/index.js
+++ b/node_modules/argos-cli/lib/index.js
@@ -51,7 +51,12 @@ _commander.default.command('upload <directory>').description('Upload screenshots
     const res = await (0, _upload.default)(_objectSpread({
       directory
     }, command));
-    json = await res.json();
+    const text = await res.text();
+    try {
+      json = JSON.parse(text)
+    } catch (error) {
+      throw new Error(`${res.status}: Failed to parse response body as JSON:\n\n${text}`)
+    }
 
     if (json.error) {
       throw new _upload.UploadError(json.error.message);
