diff --git a/node_modules/jsdom/lib/jsdom/living/nodes/HTMLOrSVGElement-impl.js b/node_modules/jsdom/lib/jsdom/living/nodes/HTMLOrSVGElement-impl.js
index 468f263..2ef0288 100644
--- a/node_modules/jsdom/lib/jsdom/living/nodes/HTMLOrSVGElement-impl.js
+++ b/node_modules/jsdom/lib/jsdom/living/nodes/HTMLOrSVGElement-impl.js
@@ -45,6 +45,10 @@ class HTMLOrSVGElementImpl {
 
     const previous = this._ownerDocument._lastFocusedElement;
 
+    if (previous === this) {
+      return;
+    }
+
     focusing.fireFocusEventWithTargetAdjustment("blur", previous, this);
     this._ownerDocument._lastFocusedElement = this;
     focusing.fireFocusEventWithTargetAdjustment("focus", this, previous);
